### PR TITLE
Optional avformat decoder

### PR DESCRIPTION
**Pull request by viric**
In my prev PR I missed this and made this parameter mandatory.

I think it should not be; it should figure out the decoder automatically, if not set.

### CODE REVIEW

File: modules/avformat/avformat.c [hunk 1]
```diff
@@ -433,10 +433,12 @@ static int module_init(void)
...
-	avformat_decoder = avcodec_find_decoder_by_name(decoder);
-	if (!avformat_decoder) {
-		warning("avformat: decoder not found (%s)\n", decoder);
-		return ENOENT;
+	if (strnlen(decoder, sizeof(decoder)) > 0) {
+		avformat_decoder = avcodec_find_decoder_by_name(decoder);
+		if (!avformat_decoder) {
+			warning("avformat: decoder not found (%s)\n", decoder);
+			return ENOENT;
+		}
```
Comment (alfredh) [inline modules/avformat/avformat.c:441]: great, could you please use `str_isset` instead, it is more efficient

File: modules/gst/gst.c [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include <re.h>
 #include <rem.h>
 #include <baresip.h>
+#include <unistd.h>
```
Comment (alfredh) [inline modules/gst/gst.c:16]: this is an unrelated change, please make a separate PR for it.
stating the problem would also be good, i.e. why the change is needed

File: modules/avformat/avformat.c [hunk 3]
```diff
@@ -303,7 +303,7 @@ int avformat_shared_alloc(struct shared **shp, const char *dev,
 		}
 	}
 
-	if (strnlen(avformat_inputformat, sizeof avformat_inputformat)	> 0) {
+	if (str_isset(avformat_inputformat)	> 0) {
```
Comment (alfredh) [inline modules/avformat/avformat.c:306]: `str_isset` returns bool. the code should be:

```c
if (str_isset(avformat_inputformat))
```

Comment (viric) [reply modules/avformat/avformat.c]: ooops
